### PR TITLE
feat(mailbox): project-scope auto-claimed pane aliases

### PR DESCRIPTION
**Pull request by sam-phinizy**
## Summary
- Auto-claimed aliases now resolve their project scope via pane → pty → session → project_id at upsert time, so the same pane name in two different projects no longer collides on a single global binding.
- Falls back to the global scope (legacy behavior) when the lookup chain is incomplete — e.g. PTY hasn't registered yet at the moment of first upsert. A subsequent upsert (rename, replacePty) re-resolves and migrates the binding cleanly.
- Closes the **project-scoping for auto-claims** item from #161.

## Stack
This PR is **the base of #163 → #164**. Review and merge this one first; downstream PRs auto-rebase onto \`main\` as their bases land.

Review order: **#162 → #163 → #164**.

## Test plan
- [x] \`cargo test --bin roux commands::panes\` — 8 new tests pass (resolver edge cases + cross-project / same-project / late-binding integration)
- [x] \`cargo test --bin roux\` — full suite, no regressions
- [x] \`cargo clippy --bin roux --tests\` — clean
- [ ] Manual: create two projects with a pane named \`reviewer\` in each, confirm both auto-claim @reviewer chips show up

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Alias auto-claims are now scoped per project, allowing the same alias name to coexist across different projects. Within a project, the first auto-claim takes effect; subsequent panes require manual alias assignment.

* **Documentation**
  * Updated mailbox troubleshooting guide to clarify alias auto-claim behavior when multiple panes share identical names.

[Review Change Stack](https://app.coderabbit.ai/change-stack/phin-tech/roux/pull/162)

<h3>Greptile Summary</h3>

- Introduces `pty_project_scope` to walk the `pty → session → project_id` chain at upsert time, so same-named panes in different projects each get their own scoped auto-claimed alias. Gracefully falls back to the global scope and re-resolves on subsequent upserts (rename, `replacePty`).
- `pty_id` is now captured from the record before the `upsert` move, eliminating a previously possible second round-trip through the pane service.
- Eight new tests cover the resolver's edge cases (missing PTY, missing session, no project), cross-project non-collision, the global→project scope migration path, and the within-project first-wins invariant.

<details><summary><h3>Confidence Score: 5/5</h3></summary>

Safe to merge — the resolution logic is sound, all fallback paths return `None` gracefully, and the migration from global to project scope is correctly handled by the cross-scope `unbind_for_pane` before re-bind.

Thoroughly traced all scope-migration edge cases (global→project, project→global, dual-race, empty pty_id sentinel, idempotent re-upsert). No P0 or P1 issues found. The two previously flagged code-quality items (`'static` bound on test lookup, redundant round-trip) are already in prior review threads; the round-trip issue is in fact fixed by this PR's `pty_id` pre-capture.

No files require special attention.
</details>

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| src-tauri/src/commands/panes.rs | Adds `pty_project_scope` async helper and integrates it into `upsert_pane_record` to resolve project-scoped alias claims; captures `pty_id` before the move to avoid a second round-trip; adds 8 well-structured unit/integration tests covering resolver edge cases and the late-binding migration scenario. |
| docs/features/mailbox.md | Troubleshooting entry updated to reflect per-project auto-claim scoping; accurately describes the new behavior where same-named panes in different projects coexist. |

</details>

</details>

<h3>Sequence Diagram</h3>

```mermaid
sequenceDiagram
    participant FE as Frontend
    participant CMD as upsert_pane_record
    participant PTY as PtyManager
    participant SES as SessionHandle
    participant ALI as AliasManager

    FE->>CMD: upsert_pane_record(record)
    CMD->>CMD: "capture pty_id = record.pty_id.clone()"
    CMD->>CMD: pane_handle.upsert(record)
    CMD->>PTY: get_info_direct(pty_id)
    alt PTY registered
        PTY-->>CMD: Some(info) → session_id
        CMD->>SES: get(session_id)
        alt "Session found & has project"
            SES-->>CMD: Some(session) → project_id
        else Session missing or no project
            SES-->>CMD: "None → project_id = None (global)"
        end
    else PTY not yet registered
        PTY-->>CMD: "None → project_id = None (global)"
    end
    CMD->>ALI: try_auto_claim_from_pane_name(pane_id, name, project_id)
    alt First upsert (no PTY)
        ALI-->>CMD: global scope claim (reviewer, None)
    else Later upsert (PTY registered)
        ALI->>ALI: unbind_for_pane (release prior scope)
        ALI->>ALI: bind under project scope
        ALI-->>CMD: project-scoped claim (reviewer, proj-A)
    end
```

<sub>Reviews (4): Last reviewed commit: ["fix: address PR #162 review findings"](https://github.com/phin-tech/roux/commit/47ada6dd7edb181f4dad829599fa034277d42f2c) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31437477)</sub>

### CODE REVIEW

File: src-tauri/src/commands/panes.rs [hunk 1]
```diff
@@ -1,6 +1,29 @@
...
+async fn pane_project_scope(
+    pane_handle: &PaneHandle,
+    session_handle: &SessionHandle,
+    pty_session_lookup: impl Fn(&str) -> Option<String>,
+    pane_id: &str,
+) -> Option<String> {
+    let records = pane_handle.list_by_ids(vec![pane_id.to_string()]).await.ok()?;
+    let record = records.into_iter().next()?;
+    let session_id = pty_session_lookup(&record.pty_id)?;
+    let session = session_handle.get(&session_id).await.ok().flatten()?;
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src-tauri/src/commands/panes.rs]: Fixed in 47ada6d. Dropped the redundant `pane_handle.list_by_ids` call — the resolver now takes pty_id directly, captured from `record.pty_id.clone()` before the upsert moves the record. Renamed to `pty_project_scope` to reflect the simpler signature.

File: src-tauri/src/commands/panes.rs [hunk 2]
```diff
@@ -39,3 +76,270 @@ pub(crate) async fn remove_pane_record(
...
+    /// Helper: build a closure over a static pty→session table so the
+    /// tests don't need a live PtyManager.
+    fn pty_lookup(table: &'static [(&'static str, &'static str)]) -> impl Fn(&str) -> Option<String> {
+        move |pty_id| {
+            table
+                .iter()
+                .find(|(p, _)| *p == pty_id)
+                .map(|(_, s)| s.to_string())
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src-tauri/src/commands/panes.rs:120]: Won't fix: the `'static` bound is intentional for now — every current test call site uses string literals, and inlining the closure would just duplicate three lines per test. Happy to revisit when integration tests start needing runtime-built tables.

File: crates/roux-core/src/models/subscription.rs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    pub id: String,
+    /// Canonical alias receiving deliveries. Need not be currently bound
+    /// to a pane — queued mail accrues for whoever claims it next.
+    pub alias: String,
+    /// Validated glob pattern. Empty / invalid patterns are rejected at
+    /// the manager boundary.
+    pub pattern: String,
+    /// Project scope. `None` = global. A subscription only matches
+    /// events whose `project_id` is `None` or equals this scope.
+    #[serde(default)]
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply crates/roux-core/src/models/subscription.rs]: Moved to #163 — this PR was split into a stack and the subscription work now lives in https://github.com/phin-tech/roux/pull/163. The doc-comment fix you flagged is a real issue and I'll address it on that PR.

File: src-tauri/src/subscriptions/manager.rs [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+            return;
+        };
+        let store = self.inner.lock().expect("subscription store poisoned");
+        if let Err(e) = save_to_path(store.entries(), path.as_path()) {
+            eprintln!(
+                "[roux] subscription persistence failed at {}: {e}",
+                path.display(),
+            );
+        }
+    }
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src-tauri/src/subscriptions/manager.rs]: Moved to #163 — bus subscriptions now live on https://github.com/phin-tech/roux/pull/163. Will evaluate the mutation-API success-reporting concern there.

File: src-tauri/src/subscriptions/persistence.rs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+pub(crate) fn save_to_path(entries: &[BusSubscription], path: &Path) -> std::io::Result<()> {
+    if let Some(parent) = path.parent() {
+        fs::create_dir_all(parent)?;
+    }
+    let envelope = SubscriptionsFile { version: SCHEMA_VERSION, data: entries.to_vec() };
+    let json = serde_json::to_string_pretty(&envelope)?;
+    fs::write(path, json)
+}
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src-tauri/src/subscriptions/persistence.rs]: Moved to #163 — subscription persistence now lives on https://github.com/phin-tech/roux/pull/163. Atomic-write concern noted; I'll handle it there.

File: src/lib/components/MailboxPanel.svelte [hunk 6]
```diff
@@ -53,7 +58,17 @@
...
+
+  // Subscription compose form state. Shown only when the Subscriptions
+  // tab is active. Pattern is validated server-side; we surface the
+  // backend error inline to keep the form concrete-feedback-driven.
+  let subAlias = $state("");
+  let subPattern = $state("");
+  let subProjectId = $state<string | null>(null);
+  let subSubmitting = $state(false);
+  let subError = $state<string | null>(null);
+  let deletingSubId = $state<string | null>(null);
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src/lib/components/MailboxPanel.svelte]: Moved to #163 — the Subscriptions tab and create form now live on https://github.com/phin-tech/roux/pull/163. Will wire project scope into the create form on that PR.

File: src-tauri/src/mailbox/manager.rs [hunk 7]
```diff
@@ -101,9 +157,10 @@ impl MailboxManager {
...
     ) -> bool {
         let now = now_ms();
+        let patterns = self.patterns_for(recipient, ProjectFilter::Any);
         let changed = {
             let mut store = self.inner.lock().expect("event store poisoned");
-            store.mark_read(event_id, recipient, now)
+            store.mark_read(event_id, recipient, &patterns, now)
         };
         if changed {
             self.persist_read_state();
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src-tauri/src/mailbox/manager.rs:109]: Moved to #163 — `MailboxManager`'s subscription wiring lives on https://github.com/phin-tech/roux/pull/163. The cross-project auth bug is real and I'll fix it there: scope the patterns to the event's project before passing to `recipient_owns`.

File: src-tauri/src/commands/panes.rs [hunk 8]
```diff
@@ -1,6 +1,29 @@
...
+    session_handle: &SessionHandle,
+    pty_session_lookup: impl Fn(&str) -> Option<String>,
+    pane_id: &str,
+) -> Option<String> {
+    let records = pane_handle.list_by_ids(vec![pane_id.to_string()]).await.ok()?;
+    let record = records.into_iter().next()?;
+    let session_id = pty_session_lookup(&record.pty_id)?;
+    let session = session_handle.get(&session_id).await.ok().flatten()?;
+    session.project_id
+}
```
[the remark this answers is not in the document]
Comment (sam-phinizy) [reply src-tauri/src/commands/panes.rs]: Fixed in 47ada6d — same finding as Copilot. Renamed to `pty_project_scope`, takes pty_id directly (cloned from record before the upsert moves it), dropped the redundant pane_handle dependency entirely.
